iconservice/icon_service_engine.py

```diff
@@ -524,7 +528,20 @@ def invoke(self,
             context.block_batch.update(context.tx_batch)
             context.tx_batch.clear()
         else:
+            tx_timer = Timer()
+            tx_timer.start()
+
             for index, tx_request in enumerate(tx_requests):
+                # Adjust the number of transactions in a block to make sure that
+                # a leader can broadcast a block candidate to validators in a specific period.
+                if is_block_editable:
```

Comment:
This function works for leader only?

Reply:
Yes. This feature works for leader only.